homeassistant/helpers/config_validation.py

```diff
@@ -387,47 +387,17 @@ def validator(value):
     return validator
 
 
-def list_or_comma_separated(member_validator=match_all) -> Callable:
-    """Return a validator.
-
-    The validator checks that the value is a list or a comma-separated string
-    that fits member_validator.
-    """
-    def validate(value: Union[str, Sequence]) -> Sequence:
-        """Validate string or sequence."""
-        if value is None:
-            raise vol.Invalid('Value can not be None')
-        if isinstance(value, str):
-            value = [member.strip() for member in value.split(',')]
-        try:
-            return [member_validator(member) for member in value]
-        except (ValueError, TypeError):
-            raise vol.Invalid(
-                'Expected sequence or string, got {}'.format(value))
-
-    return validate
-
-
-def validate_some_keys(keys: Sequence[str], validator: Callable) -> Callable:
-    """Return a validator that validates specified dictionary keys."""
-    def validate(value: Dict) -> Dict:
-        """Validate some of the dictionary keys."""
-        if not isinstance(value, dict):
-            raise vol.Invalid('Value {} is not a dictionary'.format(value))
-        result = OrderedDict()
-        for key, val in value.items():
-            if key in keys:
-                result[key] = validator(val)
-            else:
-                result[key] = val
-        return result
-
-    return validate
+def ensure_list_csv(value: Any) -> Sequence:
+    """Ensure that input is a list or make one from comma-separated string."""
+    if value is None:
+        raise vol.Invalid('Value can not be None')
```

Comment:
Do not check for `None` here, rather let that fall through and check for a minimum length in the array, for example:

```python
vol.All(cv.ensure_list_csv, vol.Length(min=1), [cv.string], [vol.Lower])
```

Otherwise you can send in an empty yaml list (not `None`) and get through, like this: `entity_id: []`